### PR TITLE
Scale SpeX spectra

### PR DESCRIPTION
**Pull request by mdcleir**
Hi,

I wrote code to calculate the scaling factor for the SpeX spectra, based on JHK photometry. It is calculating the mean of the 3 factors, and writing it to the .dat file. If there is already a spex correction factor in the .dat file, it will be overwritten. The way in which I coded the latter looks a bit cumbersome to me, so let me know if you have an idea to make this part better.

The "NaN" problem in the sum is not a problem anymore as StarData automatically replaces NaN values in the spectrum by 0s when reading in the spectrum.

I will add more stuff to do the scaling of the SXD and LXD separately, but need to "split" the spectra first.
Any suggestions/corrections are very welcome!

Marjorie

### CODE REVIEW

File: measure_extinction/stardata.py [hunk 1]
```diff
@@ -728,8 +736,8 @@ def read_irs(self, line, path="./", corfac=None):
         # self.fluxes *= mfac
         # self.uncs *= mfac
 
-        # correct the IRS spectra if corfacs defined
-        if "IRS" in corfac.keys():
+        # correct the IRS spectra if desired and if corfacs are defined
+        if use_corfac==True and "IRS" in corfac.keys():
```
Comment (karllark) [inline measure_extinction/stardata.py]: This should just be `if use_corfac and "IRS" in corfac_keys():` as use_corfac is already a boolean.

**Comment by karllark** [comment]
As for splitting SXD and LXD.  What about saving the two as separate files and then having two corfac_spex keywords?  Say corfac_spex_sxd and corfac_spex_lxd or something like that?  Then you could just give the merge_spex code both spectra after correction and then they will be merged into a single spectrum properly corrected?

File: measure_extinction/stardata.py (hunk 1 above)
Comment (mdcleir) [reply measure_extinction/stardata.py]: Right! Thanks!

**Comment by mdcleir** [comment]
> As for splitting SXD and LXD. What about saving the two as separate files and then having two corfac_spex keywords? Say corfac_spex_sxd and corfac_spex_lxd or something like that? Then you could just give the merge_spex code both spectra after correction and then they will be merged into a single spectrum properly corrected?

That was my plan in fact ;-) I just want to go over the final reduction steps with spextool again to try to get rid of these vertical lines before merging.

**Comment by mdcleir** [comment]
I fixed the conflicts, I think.

**Comment by karllark** [comment]
Is this PR ready to merge?  Or were you planning on implementing the split SXD and LXD in this PR?

**Comment by mdcleir** [comment]
> Is this PR ready to merge? Or were you planning on implementing the split SXD and LXD in this PR?

The code should work as it is now, for the full SpeX spectra, so I think you can merge. I am definitely planning to split the S and LXD, but can do that in a new branch. What do you think?

**Comment by karllark** [comment]
Sounds like a plan.  Merging.

**Pull request merged**